modules/file_module.php

```diff
@@ -29,13 +29,22 @@ function file_getCampus($campus, $faculty) {
 }
 
 function file_getSubject($campus, $faculty, $subject) {
+    $subjects = file_getCampus($campus, $faculty);
+    $subjects = json_decode($subjects, true);
+    $path = '';
+
+    foreach ($subjects as $s) {
+        if ($s['subject'] == $subject) {
+            $path = $s['path'];
+        }
+    }
 
 		$filename = empty($faculty) ?
     		'./cache/' . $_POST['campus'] . '-' . $_POST['subject'] . '.dat' :
     		'./cache/' . $_POST['campus'] . '-' . $_POST['faculty'] . '-' . $_POST['subject'] . '.dat';
 
     if (!file_exists($filename) || getFileOld($filename) > CACHE_TIMELEFT)
-        file_put_contents($filename, icress_getSubject($campus, $faculty, $subject));
+        file_put_contents($filename, icress_getSubject($path));
```

Comment:
The current URL looks like this:
`https://simsweb4.uitm.edu.my/estudent/class_timetable/index_tt.cfm?id1=7620E16BD398AFF4AACE29E5A81A918EBCC6&id2=701DDF471CA4AC4BEF0CC3DD505EA4671387`

There are `id1` and `id2` where I am not sure what it is. Maybe hashed value of `campus` and `subject`. Who knows.

That's why I just dump the whole path here